cmd/ctl/agent.go

```diff
@@ -434,7 +436,19 @@ func clusterSecretName(agentName string) string {
 func serverURL(address, agentName string) (string, error) {
 	_, err := netip.ParseAddrPort(address)
 	if err != nil {
-		return "", err
+		// We now have a host:port address, so we need to validate the host
+		tok := strings.SplitN(address, ":", 2)
+		if len(tok) != 2 {
+			return "", fmt.Errorf("invalid address: %s", address)
+		}
+		addr := tok[0]
+		port := tok[1]
+		if len(validation.NameIsDNSLabel(addr, false)) > 0 {
```

Comment:
This returns an error for hostnames with a `.` dot. e.g. `example.com`. Should we use `validation.IsDNS1123Subdomain` instead?